sale_order_type/tests/test_sale_order_type.py

```diff
@@ -287,3 +287,74 @@ def test_sale_order_type_not_required(self):
             order_line.product_uom_qty = 1.0
         sale_form.type_id = self.sale_type.browse()
         sale_form.save()
+
+    def test_credit_note_preserves_sale_type_from_sale_order(self):
+        """Test credit notes preserve sale order type.
+
+        When creating a credit note (refund) from an invoice that originated
+        from a sale order, the sale_type_id from the sale order should be
+        maintained and not overridden by the partner's default sale type.
+        """
+        # Setup: Create two different sale order types
```

Comment:
This comment is not adding too much value and even it's incorrect, as you are not creating anything.